crates/levm/src/vm.rs

```diff
@@ -47,3 +96,156 @@ impl VM {
         self.pc
     }
 }
+
+#[derive(Debug, Clone, Default)]
+pub struct Memory {
+    data: Vec<u8>,
+}
+
+impl Memory {
+    pub fn new() -> Self {
+        Self { data: Vec::new() }
+    }
+
+    pub fn resize(&mut self, offset: usize) {
+        if offset + 32 > self.data.len() {
+            self.data.resize(offset + 32, 0);
+        }
+    }
+
+    pub fn load(&self, offset: usize) -> U256 {
+        let mut value_bytes = [0u8; 32];
+        value_bytes.copy_from_slice(&self.data[offset..offset + 32]);
+        U256::from(value_bytes)
+    }
+
+    pub fn store_bytes(&mut self, offset: usize, value: &[u8]) {
+        self.data
+            .splice(offset..offset + value.len(), value.iter().copied());
+    }
+
+    pub fn size(&self) -> U256 {
+        U256::from(self.data.len())
+    }
+
+    pub fn copy(&mut self, src_offset: usize, dest_offset: usize, size: usize) {
+        let mut temp = vec![0u8; size];
+
+        temp.copy_from_slice(&self.data[src_offset..src_offset + size]);
+
+        self.data[dest_offset..dest_offset + size].copy_from_slice(&temp);
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Let's move the tests to `crates/levm/tests/`.

Reply:
done!